src/treecons.ml

```diff
@@ -0,0 +1,138 @@
+(** Tree constructors: a simpler representation of terms to be used by a
+    decision tree. *)
+
+open Extra
+
+(** A constructor is the representation of a symbol along with the number of
+    arguments to which it is applied. *)
+type treecons =
+  | Symb of
+      { c_sym : Files.module_path
+      (** Fully qualified name of the constructor. *)
+      ; c_ari : int
+      (** Arity of the considered constructor.  A same symbol representation
+          may generate several constructors with different arities.  See
+          [tests/OK/arity_diff.lp] for an example. *) }
```

Comment:
This iexample is not satisfactory. In this example, you have a defined function symbol occurring at the *head* of two LHS applied to a different number of arguments. But, if I understand well, treecons are used for symbols occurring *inside* the arguments of the head symbol of a LHS (and, indeed, this head symbol does not occur in the type of rules). Instead, I am looking for an example like f a (a b) --> r. Thanks to add the example explicitly in the comment.

Reply:
Ah yes, I'll add the example to the file as well

Reply:
Perhaps, the point is to simply indicate also that the arity is the number of subtrees?